moment.js

```diff
@@ -2033,7 +2058,46 @@
         }
     }
 
-    // date from iso format or fallback
+    // date from ietf format
+    function configFromIETF(config) {
+        var dateString, match, dateCheck, stDay;
+
+        dateString = config._i.
+        	replace(/(?:\([^\)]*\))/g, ' ').	// Replace comments with a single space
+        	replace(/[\l\n\r\t]/g, ' ').		// Replace 'fold' characters with a single space	
+        	replace(/\s{2,}/g, ' ').			// Replace multiple spaces with a single space
+        	replace(/(?:^\s|\s$)/g, '');		// Remove leading and training spaces	
+		match = dateString.match(ietfRegex);
+
+        if (match) {
+            getParsingFlags(config).ietf = true;
+            
+            dateCheck = new Date( match[2] + " GMT");
+            if (!dateCheck) {
+            	config._isValid = false;
+            	return;
+            }
+            stDay = defaultLocaleWeekdaysShort[dateCheck.getDay()];
+			if (!match[1]) { 					// Add Day of week when missing
+				match[1] = stDay + ", ";
+			} 
+			else {								// Otherwise check Day of week matches
+				if (stDay !== match[1].replace(/,?\s$/,'')) {
+	            	config._isValid = false;
+	            	return;
+				}
+			}
+			if (!match[3]) { match[3] = ":00";}	// Add default seconds when missing
```

Comment:
Why do you need to add the default seconds? They should be zero by default already.